### PR TITLE
migrate to hyper v1

### PR DESCRIPTION
**Pull request by ahl**
There's going to need to be some coordination with these:
- https://github.com/oxidecomputer/dropshot/pull/1028
- https://github.com/oxidecomputer/progenitor/pull/883
- https://github.com/oxidecomputer/crucible/pull/1428
- https://github.com/oxidecomputer/maghemite/pull/344
- https://github.com/oxidecomputer/propolis/pull/745

I hope to merge this and then get crucible, propolis, and maghemite depending on this updated version so that we can return the reqwest 0.11 dependency over there.

After merging this I intend to temporarily depend on a maghemite branch (see https://github.com/oxidecomputer/maghemite/pull/378) to navigate past the omicron-common test failure.

### CODE REVIEW

File: .cargo/xtask.toml [hunk 1]
```diff
@@ -13,6 +13,7 @@
 [libraries."libgcc_s.so.1"]
 [libraries."libipcc.so.1"]
 [libraries."libkstat.so.1"]
+[libraries."liblzma.so.5"]
```
Comment (ahl) [inline .cargo/xtask.toml:16]: @papertigers can I just... add this? @rcgoodfellow this is due to a new dep from falcon from https://github.com/oxidecomputer/falcon/pull/88

File: nexus/tests/integration_tests/certificates.rs [hunk 2]
```diff
@@ -599,11 +600,7 @@ async fn test_silo_certificates() {
         );
     if let oxide_client::Error::CommunicationError(error) = error {
         assert!(error.is_connect());
-        assert!(
-            error.to_string().contains("invalid peer certificate")
-                || error.to_string().contains("self-signed certificate")
-                || error.to_string().contains("self signed certificate")
-        );
+        assert!(error.chain().to_string().contains("self-signed certificate"));
```
Comment (ahl) [inline nexus/tests/integration_tests/certificates.rs:603]: @jmpesp I think this was you; does this new check suffice?

File: .cargo/xtask.toml (hunk 1 above)
Comment (papertigers) [reply .cargo/xtask.toml:16]: That's an explicit allow list, so as long as we expect that lib to be present in the gz and ngz it's the right place to stick it. If it's a lib we don't want to leak out of a single binary we can add it an allow list similar to what libipcc has.

File: Cargo.toml [hunk 3]
```diff
@@ -505,11 +507,11 @@ pretty_assertions = "1.4.1"
 pretty-hex = "0.4.1"
 prettyplease = { version = "0.2.22", features = ["verbatim"] }
 proc-macro2 = "1.0"
-progenitor = { git = "https://github.com/oxidecomputer/progenitor", branch = "main" }
-progenitor-client = { git = "https://github.com/oxidecomputer/progenitor", branch = "main" }
-bhyve_api = { git = "https://github.com/oxidecomputer/propolis", rev = "24a74d0c76b6a63961ecef76acb1516b6e66c5c9" }
-propolis-client = { git = "https://github.com/oxidecomputer/propolis", rev = "a9c5fdac442fa6fd5a65e413ea874fe3c62c3fd9" }
-propolis-mock-server = { git = "https://github.com/oxidecomputer/propolis", rev = "a9c5fdac442fa6fd5a65e413ea874fe3c62c3fd9" }
+progenitor = "0.8.0"
```
Comment (bnaecker) [inline Cargo.toml:510]: This is the best part!

File: Cargo.toml [hunk 4]
```diff
@@ -524,7 +526,8 @@ reedline = "0.33.0"
 ref-cast = "1.0"
 regex = "1.10.6"
 regress = "0.9.1"
-reqwest = { version = "0.11", default-features = false }
+reqwest = { version = "0.12", default-features = false }
+reqwest11 = { package = "reqwest", version = "0.11", default-features = false }
```
Comment (bnaecker) [inline Cargo.toml:530]: Just a sanity check that this will go away when we merge Maghemite back in here?

File: internal-dns/src/resolver.rs [hunk 5]
```diff
@@ -834,8 +833,8 @@ mod test {
         // The DNS server is running, but has no records. Expect a failure.
         let err = client.test_endpoint().await.unwrap_err();
         assert!(
-            err.to_string().contains("no record found"),
-            "Unexpected Error (expected 'no record found'): {err}",
+            err.to_string().contains("error sending request"),
```
Comment (bnaecker) [inline internal-dns/src/resolver.rs:836]: Why did this bit change? If the server is running, but doesn't have any records, shouldn't the "send part" succeed?

**Review by bnaecker — COMMENTED** [review]
Looks alright to me!

File: Cargo.toml (hunk 4 above)
Comment (ahl) [reply Cargo.toml:530]: That's right. Need to get everyone updated and pull out the reqwest 0.11 weeds

File: internal-dns/src/resolver.rs (hunk 5 above)
Comment (ahl) [reply internal-dns/src/resolver.rs:836]: I think I can improve this check (insofar as it's useful). The part we were looking for is just farther down the nested chain of errors. I'll do that in a follow on

File: nexus/tests/integration_tests/certificates.rs (hunk 2 above)
Comment (jmpesp) [reply nexus/tests/integration_tests/certificates.rs:603]: It's been so long I don't remember doing this haha!

I'm not sure this check is right: it looks like this part of the test is meant to check that a client can't connect using the wrong certificate, and this check is asserting it won't trust a self-signed certificate. That's not quite the same thing. I think we need either [`add_root_certificate`](https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.add_root_certificate) or [`danger_accept_invalid_certs`](https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.danger_accept_invalid_certs) in order to test that the client accepts the self-signed cert but fails due to the incorrect certificate being used.

Comment (ahl) [reply nexus/tests/integration_tests/certificates.rs:603]: https://github.com/oxidecomputer/omicron/issues/6716

File: internal-dns/src/resolver.rs (hunk 5 above)
Comment (ahl) [reply internal-dns/src/resolver.rs:836]: https://github.com/oxidecomputer/omicron/issues/6717

**Pull request merged**